guides/source/caching_with_rails.md

```diff
@@ -512,13 +512,19 @@ class ProductsController < ApplicationController
 end
 ```
 
-### A note on weak ETags
-
-Etags generated by Rails are weak by default. Weak etags allow semantically equivalent responses to have the same etags, even if their bodies do not match exactly. This is useful when we don't want the page to be regenerated for minor changes in response body. If you absolutely need to generate a strong etag, it can be assigned to the header directly.
-
-```ruby
-  response.add_header "ETag", Digest::MD5.hexdigest(response.body)
-```
+### Notes on strong and weak ETags
+Strong ETag sets a "strong" ETag validator on the response. Strong ETag implies that
+response should be exactly same and byte by byte identical. Useful when doing Range
+requests within a large video or PDF file. Some CDNs support only strong ETags,
+like Akamai.
+
+ETag and weak ETag, they both set "weak" ETag validator on the response. A weak
+ETag indicates semantic equivalent but not byte-for-byte identical. These are
+perfect for browser caching of HTML pages where we don't care about exact
+equality. Etags generated by Rails are weak by default.
+
+Quotes (") are the significant part of the Etag. A quoted ETag and an unquoted
+one are not the same entity.
```

Comment:
This isn't really relevant to users since they aren't ever responsible for quoting. It may confuse people into thinking it's their responsibility.